contracts/contracts/eosio.system/src/voting.cpp

```diff
@@ -86,6 +86,10 @@ namespace eosiosystem {
    void system_contract::unregprod( const name& producer ) {
       require_auth( producer );
 
+      const auto voter = _voters.find( producer.value );
+      check( voter != _voters.end(), "producer is not found in voter table" );
+      check( voter->vote_mature_time <= current_time_point(), "producers are not allowed to unreg during stake lock period" );
```

Comment:
let's use full words
"producers are not allowed to unregister during stake lock period"